schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/ConfigEntry.java

```diff
@@ -128,6 +137,44 @@ public String toString() {
                 '}';
     }
 
+    /**
+     * Parses a String to a typed value
+     *
+     * @param <T> type of the resulting value.
+     */
+    public interface Converter<T> {
+        T convert(String str);
+    }
+
+    public static class IntegerConverter implements Converter<Integer> {
+
+        private static Converter<Integer> instance = new IntegerConverter();
+
+        @Override
+        public Integer convert(String str) {
+            return Integer.valueOf(str);
```

Comment:
valueOf can throw NumberFormatException, may be we should wrap this exception with SchemaRegistry custom exception like ConfigTypeConvertionException ?

Reply:
We might do this, but what would be the added value of doing so? We would just exchange one type of exception to an another.

Reply:
The idea is that ConfigTypeConvertionException - or something with a better name - would act as a root exception class for Converters, meaning that it would not only wrap NumberFormatException mentioned here but it could be used for other exceptions like in the IntegerConverter I have commented on before. By having a single root exception class it would be easier to debug where the issue is coming from.